Sources/RemoteMessaging/RemoteMessagingStore.swift

```diff
@@ -0,0 +1,383 @@
+//
+//  RemoteMessagingStore.swift
+//
+//  Copyright © 2022 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+import CoreData
+import BrowserServicesKit
+import Persistence
+
+public enum RemoteMessagingStoreError: Error {
+    case saveConfigFailed
+    case invalidateConfigFailed
+    case updateMessageShownFailed
+    case saveMessageFailed
+    case updateMessageStatusFailed
+    case deleteScheduledMessageFailed
+}
```

Comment:
`Pixel.fire` calls have been replaced with `EventMapping` mechanism (also commonly used in BSK), and now whenever a pixel was fired, it's replaced with the event mapping emitting a respective event with an associated error parameter. Client app provides EventMapping class that receives the events and fires pixels.